lib/route-data.js

```diff
@@ -8,7 +8,7 @@ class RouteData {
    * @param {!Array<string>=} namedParameters list in camelCase. Will be
    *     converted to a map of camelCase and hyphenated.
    * @param {boolean=} requiresAuthentication
-   * @param {function():(Promise<unknown>|undefined)=} beforeEnter
+   * @param {function():Promise=} beforeEnter
```

Comment:
You could try `Promise<?>` - that's how you indicate `unknown`. But you are right - it doesn't really matter.